server/server.go

```diff
@@ -324,8 +320,8 @@ func (s *Server) Context() context.Context {
 	return s.serverLoopCtx
 }
 
-func (s *Server) startServerLoop() {
-	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(context.Background())
+func (s *Server) startServerLoop(ctx context.Context) {
+	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(ctx)
 	s.serverLoopWg.Add(3)
 	go s.leaderLoop()
```

Comment:
do we need add ctx for them?

Reply:
These three goroutines have their own contexts based on `serverLoopCtx`